File: app/src/main/java/linusfessler/alarmtiles/stopwatch/StopwatchDao.java
package linusfessler.alarmtiles.stopwatch;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import io.reactivex.Observable;

@Dao
public interface StopwatchDao {

    @Query("SELECT * FROM Stopwatch")
    Observable<Stopwatch> select();

    @Insert
    void insert(Stopwatch stopwatch);

    @Update
    void update(Stopwatch stopwatch);

    @Query("SELECT COUNT(*) FROM Stopwatch")
    int count();
}

File: app/src/main/java/linusfessler/alarmtiles/tiles/stopwatch/StopwatchDao.kt
package linusfessler.alarmtiles.tiles.stopwatch

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import io.reactivex.Observable

@Dao
interface StopwatchDao {
    @Query("SELECT * FROM Stopwatch")
    fun select(): Observable<Stopwatch>

    @Insert
    fun insert(stopwatch: Stopwatch)

    @Update
    fun update(stopwatch: Stopwatch)

    @Query("SELECT COUNT(*) FROM Stopwatch")
    fun count(): Int
}